vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1122,29 +1136,49 @@ func resourceVSphereVirtualMachineImport(d *schema.ResourceData, meta interface{
 		return nil, fmt.Errorf("VM %q is a template and cannot be imported", name)
 	}
 
-	// Quickly walk the SCSI bus and determine the number of contiguous
-	// controllers starting from bus number 0. This becomes the current SCSI
+	// Quickly walk the storage busses and determine the number of contiguous
+	// controllers starting from bus number 0. This becomes the current
 	// controller count. Anything past this is managed by config.
-	log.Printf("[DEBUG] Determining number of SCSI controllers for VM %q", name)
+	log.Printf("[DEBUG] Determining number of controllers for VM %q", name)
 	scsiBus := make([]bool, 4)
+	sataBus := make([]bool, 4)
+	ideBus := make([]bool, 2)
 	for _, device := range props.Config.Hardware.Device {
-		sc, ok := device.(types.BaseVirtualSCSIController)
-		if !ok {
-			continue
+		switch dev := device.(type) {
+		case types.BaseVirtualSCSIController:
+			scsiBus[dev.GetVirtualSCSIController().BusNumber] = true
+		case types.BaseVirtualSATAController:
+			sataBus[dev.GetVirtualSATAController().BusNumber] = true
+		case *types.VirtualIDEController:
+			ideBus[dev.GetVirtualController().BusNumber] = true
 		}
-		scsiBus[sc.GetVirtualSCSIController().BusNumber] = true
 	}
-	var ctlrCnt int
+	var scsiCtlrCnt int
```

Comment:
This loop for each bus list could be converted into a small function, since all three lists are of the same type

Reply:
That does look much better.